### PR TITLE
Change win checking method (FinishedGameState now means that the game…

### PR DESCRIPTION
**Pull request by agawda**
… was won, it gives information which player is the winner), add GameWinner enum (for checking who won), add tests to new classes, refactor classes that were using exceptions to check if the game was won

### CODE REVIEW

File: client/src/main/java/com/blackship/battlesheep/fx/BoardViewUpdater.java [hunk 1]
```diff
@@ -55,35 +55,46 @@ public BoardViewUpdater build() {
...
         updatePlayerBoard(salvoHandler.getPlayerHitPositions());
         enemyMastPositions
                 .stream()
-                .filter(data -> !data.isDisabled())
-                .forEach(data -> data.setStyle(ButtonUtils.defaultButtonColorStyle()));
+                .filter(data -> enemyHitPositions.contains(Integer.parseInt(data.getId())))
+                .forEach(data -> {
+                    data.setDisable(false);
+                    data.setStyle(ButtonUtils.defaultButtonColorStyle());
+                });
```
Comment (Dzionek95) [inline client/src/main/java/com/blackship/battlesheep/fx/BoardViewUpdater.java]: 👍

File: client/src/main/java/com/blackship/battlesheep/fx/BoardViewUpdater.java [hunk 2]
```diff
@@ -55,35 +55,46 @@ public BoardViewUpdater build() {
...
+            fireCannons(loadedCannons);
         }
     }
+
+    private void fireCannons(List<Integer> cannons) {
+        salvoHandler.fireSalvoCannonade(cannons);
+        loadedCannons.clear();
+        boardViewUpdaterListener.update("...!!!!FIRE!!!!...\n");
+    }
     //TODO Solve magic number bug(?)
```
Comment (Dzionek95) [inline client/src/main/java/com/blackship/battlesheep/fx/BoardViewUpdater.java]: 👎

File: client/src/test/java/com/blackship/battlesheep/fx/utils/ButtonUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    public void setupJavaFx() {
+        final CountDownLatch latch = new CountDownLatch(1);
+        SwingUtilities.invokeLater(() -> {
+            new JFXPanel(); // initializes JavaFX environment
+            latch.countDown();
+        });
+    }
+
+
+    @DataProvider
```
Comment (Dzionek95) [inline client/src/test/java/com/blackship/battlesheep/fx/utils/ButtonUtilsTest.java]: 💛

File: client/src/test/java/com/blackship/battlesheep/fx/utils/ButtonUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            };
+    }
+
+    @Test(dataProvider = "buttonCreationData")
+    public void givenIntDataExpectNotNullButtonTest(int expectedValue, int expectedHeight, int expectedWidth) {
+        //when
+        Button button = ButtonUtils.createStyledButton(expectedValue);
+        //then
+        SoftAssert softAssert = new SoftAssert();
+//        assertEquals(button.getId(), String.valueOf(expectedValue));
```
Comment (Dzionek95) [inline client/src/test/java/com/blackship/battlesheep/fx/utils/ButtonUtilsTest.java]: why this comment?

File: server/src/main/java/com/blackship/battlesheep/communication/network/ServerCommunicationHandler.java [hunk 5]
```diff
@@ -43,18 +43,14 @@ public ServerCommunicationHandler acceptClients() throws IOException {
...
-    public void echo() throws IOException, ClassNotFoundException {
+    public void echo() throws IOException, ClassNotFoundException, WrongStateException {
         ClientSocketHandler firstClient = clients.get(0);
         ClientSocketHandler secondClient = clients.get(1);
 
         Game game = new Game();
         game.startGame(FleetGenerator.hardcodeShips(), FleetGenerator.hardcodeShips());
 
-        while(runningLoop) {
+        while (true) {
```
Comment (Dzionek95) [inline server/src/main/java/com/blackship/battlesheep/communication/network/ServerCommunicationHandler.java]: untilWeHaveFleet()?

**Review by Dzionek95 — APPROVED** [review]

File: client/src/main/java/com/blackship/battlesheep/fx/BoardViewUpdater.java (hunk 1 above)
Comment (bartwoj) [reply client/src/main/java/com/blackship/battlesheep/fx/BoardViewUpdater.java]: 👍

File: client/src/test/java/com/blackship/battlesheep/fx/utils/ButtonUtilsTest.java (hunk 3 above)
Comment (bartwoj) [reply client/src/test/java/com/blackship/battlesheep/fx/utils/ButtonUtilsTest.java]: 👍

File: client/src/test/java/com/blackship/battlesheep/fx/utils/ButtonUtilsTest.java (hunk 4 above)
Comment (bartwoj) [reply client/src/test/java/com/blackship/battlesheep/fx/utils/ButtonUtilsTest.java]: exaclty?

File: server/src/main/java/com/blackship/battlesheep/AppServer.java [hunk 6]
```diff
@@ -11,7 +12,7 @@
  */
 public class AppServer {
 
-    public static void main(String[] args) throws IOException, ClassNotFoundException {
+    public static void main(String[] args) throws IOException, ClassNotFoundException, WrongStateException {
```
Comment (bartwoj) [inline server/src/main/java/com/blackship/battlesheep/AppServer.java]: Your own exception 👍

File: server/src/main/java/com/blackship/battlesheep/communication/network/ServerCommunicationHandler.java [hunk 7]
```diff
@@ -43,18 +43,14 @@ public ServerCommunicationHandler acceptClients() throws IOException {
...
```
Comment (bartwoj) [inline server/src/main/java/com/blackship/battlesheep/communication/network/ServerCommunicationHandler.java]: Why "true"?

File: server/src/main/java/com/blackship/battlesheep/game/state/GameState.java [hunk 8]
```diff
@@ -1,27 +1,26 @@
...
 /**
  * @author Anna Gawda
  * @since 26.07.2017
  *
- * A state machine that manages the game progress. It keeps one board (in the future it will
- * keep boards for both players). It is possible to get state of the board as a String
- * and to change the state of the game. The states: starting, in progress and finished.
+ * A state machine that manages the game progress. It keeps the fleets of both players. It allows the server
+ * to make a move and to monitor the game state which can be Starting and InProgress. Making moves is only
+ * possible in the InProgress state.
```
Comment (bartwoj) [inline server/src/main/java/com/blackship/battlesheep/game/state/GameState.java:13]: "It allows the server to make a move" ?? O_O

**Review by bartwoj — APPROVED** [review]

**Pull request merged**